src/hidpp-generic.c

```diff
@@ -306,6 +306,33 @@ hidpp_read_response(struct hidpp_device *dev, uint8_t *buf, size_t size)
 	return rc >= 0 ? rc : -errno;
 }
 
+void
+hidpp_get_supported_report_types(struct hidpp_device *dev)
+{
+	if (!dev->hid_reports) {
+		hidpp_log_debug(dev, "hidpp: we don't have information about the hid reports, ignoring checks\n");
+		return;
+	}
+
+	/* reset the bits we are gonna check */
+	dev->supported_report_types &= (0xff & ~HIDPP_REPORT_SHORT);
+	dev->supported_report_types &= (0xff & ~HIDPP_REPORT_LONG);
+
+	for (unsigned i = 0; i < dev->num_hid_reports; i++)
+		if (dev->hid_reports[i].usage_page >= 0xff00 &&
+		    dev->hid_reports[i].usage_page <= 0xffff) /* vendor defined usage page (0xff00-0xffff) */
```

Comment:
nitpick: `(dev->hid_reports[i].usage_page & 0xff00) == 0xff00`

Reply:
I was going to do that but I didn't because I thought this would be a little more readable. I am going to change it.